### ISSUE TITLE
Web/UI - Build Event do not show on UI, keeps showing loading

### ISSUE THREAD

**Issue by asantos-fuze** [concourse/concourse#3563]
## Bug Report

Build Event do not show on UI, keeps showing loading.
If I list events on fly, they show up.

## Steps to Reproduce

* Deploy concourse web in one instance
* Deploy concourse worker in another instance
* Postgres on RDS

Run pipeline/job

pipeline:
https://gist.github.com/asantos-fuze/2e7791ffa792439d54f5c2bd444aec9f
task:
https://gist.github.com/asantos-fuze/86cbca626702a62a3f935fb0542ca5f3

```
fly -t tutorial set-pipeline -c pipeline.yml -p hello-world
fly -t tutorial unpause-pipeline -p hello-world
fly -t tutorial unpause-job --job hello-world/job-hello-world
```

## Expected Results

Expected to see the output of the build on the Web/UI interface after they are finished.

## Actual Results

Web/UI, does not show Build results. Shows "Loading" endlessly
![Screenshot at Mar 21 14-07-20](https://user-images.githubusercontent.com/32873313/54757778-d59c6780-4be2-11e9-84aa-4dfa87ead92f.png)

Fly shows all the logs from the build.

```
fly -t tutorial watch hello-world/job-hello-world
initializing
waiting for docker to come up...
Pulling busybox@sha256:061ca9704a714ee3e8b80523ec720c64f6209ad3f97c0ff7cb9ec7d19f15149f...
sha256:061ca9704a714ee3e8b80523ec720c64f6209ad3f97c0ff7cb9ec7d19f15149f: Pulling from library/busybox
697743189b6d: Pulling fs layer
697743189b6d: Download complete
697743189b6d: Pull complete
Digest: sha256:061ca9704a714ee3e8b80523ec720c64f6209ad3f97c0ff7cb9ec7d19f15149f
Status: Downloaded newer image for busybox@sha256:061ca9704a714ee3e8b80523ec720c64f6209ad3f97c0ff7cb9ec7d19f15149f

Successfully pulled busybox@sha256:061ca9704a714ee3e8b80523ec720c64f6209ad3f97c0ff7cb9ec7d19f15149f.

running echo hello world
hello world
succeeded
```

## Additional Context

If I stop concourse web node, the page loads immediately.
This is because the browser receives a TCP FIN,ACK coming from the web node, and the page renders. Is this related to the fact that the `GET /api/v1/builds/21/events HTTP/1.1` has the `Accept: text/event-stream` and the events do not load until the server closes the stream?

There is no LB between the instances or reverse proxy.

## Version Info

* Concourse version: v5.0.0
* Deployment type (BOSH/Docker/binary):binary
* Infrastructure/IaaS: AWS
* Browser (if applicable): Google Chrome | 72.0.3626.121
* Did this used to work? First try on concourse

**Comment by pivotal-jamie-klassen** [comment]
@asantos-fuze do you see any messages in the Javascript console of Chrome?

**Comment by asantos-fuze** [comment]
Hi @pivotal-jamie-klassen I'm logged in.
Bellow is what I get on the browser console

![Screenshot at Mar 21 14-37-03](https://user-images.githubusercontent.com/32873313/54759995-19916b80-4be7-11e9-861f-0c9a3e35230c.png)

![Screenshot at Mar 21 14-37-29](https://user-images.githubusercontent.com/32873313/54760006-1dbd8900-4be7-11e9-9459-d80acb1c76e7.png)

**Comment by asantos-fuze** [comment]
@pivotal-jamie-klassen I was able to put concourse to work.
I received a tip on the discordapp that it might be sophos. Guess what, i had sophos running.

To overcame the issue I have set up an ELB with HTTPS in from of the concourse web and things have started to work.

**Comment by gerhard** [comment]
I had the same issue and came across this. While the root cause for my issue was related to running Concourse behind Cloudflare, I thought that it may be useful to share in case someone else comes across it.

If proxying via Cloudflare, ensure to **Disable performance**, otherwise build logs will not be sent via SSE.

This is what our page rule looks like, as configured via Terraform:

```hcl
resource "cloudflare_page_rule" "enable_sse_for_build_logs_streaming" {
  zone_id = "xxx"
  target = "cd.rabbitmq.com/*"
  actions = {
    always_use_https = "true"
    automatic_https_rewrites = "on"
    cache_level = "bypass"
    disable_performance = "true"
  }
}
```

**Comment by Cheezzhead** [comment]
@gerhard This may be a shot in the dark, but I'm facing the exact same problem. My issue is also related to running Concourse CI behind Cloudflare, evidenced by the fact that when I disable the proxy status for the concourse subdomain, it immediately works.

I have added a page rule for the subdomain for 'Cache level: Bypass' and 'Disable Performance: True', but the issue persists. Is there anything else you have had to disable (or enable) in cloudflare to remedy this?

**Comment by gerhard** [comment]
No, that's it on the Cloudflare side. The only other config is to enforce HTTPS in Cloudflare.

If there is a load balancer / proxy in front of concourse web, this also needs to handle SSE (Server Sent Events) correctly.

**Comment by taylorsilva** [comment]
# Issue Bankruptcy

This issue is being closed as part of 2025's **Issue Bankruptcy** declaration. [See here for details](https://blog.concourse-ci.org/posts/2025-01-06-concourse-2024-review/)

If you think this issue still has merit then feel free to open a new issue and reference this one.
